lib/vmdb/settings.rb

```diff
@@ -109,21 +109,33 @@ def self.build_without_local(resource)
     end
     private_class_method :build_without_local
 
+    def self.template_roots
+      roots = [Rails.root.join('config')]
+      provider_engines = Provider.subclasses.find_all(&:engine?)
```

Comment:
I like going through `Provider.subclasses` because it also forces us to have all providers subclass this :smile: 
Also it feels natural.

I'm not sure about the `engine?` thing here. Maybe `Provider.root` should return the path like `Rails.root` ?